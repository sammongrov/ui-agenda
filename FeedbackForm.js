import React, { Component } from 'react';
import { ScrollView, TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types';
import { Actions } from 'react-native-router-flux';
import { Field, reduxForm } from 'redux-form';
import { Colors } from '@ui/theme_default';
import { NavBar, Screen, Icon } from '@ui/components';
import { styles } from 'react-native-theme';
import TextField from '../../../src/idm/ui/components/form/TextField';
import SubmitButton from '../../../src/idm/ui/components/form/SubmitButton';
import {
  isEmailValid,
  isNameValid,
  // isPhoneNumberValid
} from '../../../src/idm/util/validator';

class FeedbackForm extends Component {
  static defaultProps = {
    error: false,
    invalid: false,
    submitting: false,
  };

  state = {
    name: '',
    email: '',
    // phone: '',
    comments: '',
  };

  onSubmit = (values) => {
    console.log('viswanth', values);
  };

  updateSize = (height) => {
    this.setState({
      height,
    });
  };

  render() {
    const { error, handleSubmit, invalid, submitting } = this.props;
    const {
      name,
      email,
      // phone,
      comments,
      height,
    } = this.state;
    const newStyle = { height };
    return (
      <Screen>
        <NavBar
          titleText="Feedback"
          leftComponent={
            <TouchableOpacity
              onPress={() => {
                if (Actions.currentScene === 'FeedbackFormScene') {
                  Actions.pop();
                }
              }}
            >
              <Icon name="chevron-left" color={Colors.NAV_ICON} size={36} />
            </TouchableOpacity>
          }
        />
        <ScrollView
          contentContainerStyle={{
            flexGrow: 1,
            paddingHorizontal: 10,
            paddingTop: 10,
            backgroundColor: '#F6F7F9',
          }}
        >
          <Field
            autoCapitalize="none"
            component={TextField}
            keyboardType="default"
            name="name"
            // onSubmitEditing={this.focusPasswordInput}
            label="Name"
            onChange={(textName) => {
              this.setState({ name: textName });
            }}
            value={name}
            validate={isNameValid}
            returnKeyType="next"
          />
          <Field
            autoCapitalize="none"
            component={TextField}
            keyboardType="email-address"
            name="email"
            // onSubmitEditing={this.focusPasswordInput}
            label="Email"
            onChange={(textMail) => this.setState({ email: textMail })}
            value={email}
            validate={isEmailValid}
            returnKeyType="next"
          />
          {/* <Field
            autoCapitalize="none"
            // autoFocus={autoFocus}
            component={TextField}
            keyboardType="phone-pad"
            name="phone"
            // onSubmitEditing={this.focusPasswordInput}
            label="Phone"
            onChange={(textPhone) => this.setState({ phone: textPhone})}
            value={phone}
            // ref={(ref) => {
            //   this.emailInput = ref;
            // }}
            validate={validation.normalizeMobile}
            // normalize={validation.normalizeMobile}
            returnKeyType="next"
            // withRef
          /> */}
          <Field
            autoCapitalize="none"
            component={TextField}
            keyboardType="default"
            name="comment"
            label="Comment"
            onChange={(textComment) => this.setState({ comments: textComment })}
            value={comments}
            returnKeyType="next"
            multiline
            onContentSizeChange={(e) => this.updateSize(e.nativeEvent.contentSize.height)}
            textAreacontainerStyle={newStyle}
            // withRef
          />
          <SubmitButton
            disabled={invalid && !error}
            loading={submitting}
            onPress={handleSubmit(this.onSubmit)}
            text="Submit"
            containerStyle={[styles.marginHorizontal18, styles.marginTop20]}
          />
        </ScrollView>
      </Screen>
    );
  }
}

FeedbackForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  error: PropTypes.bool,
  invalid: PropTypes.bool,
  submitting: PropTypes.bool,
};

export default reduxForm({
  form: 'FeedbackForm', // a unique identifier for this form
})(FeedbackForm);

// example working correctly

// import React from 'react'
// import {
//   StyleSheet,
//   Text,
//   TextInput,
//   TouchableOpacity,
//   View
// } from 'react-native'
// import { reduxForm, Field } from 'redux-form'

// const styles = StyleSheet.create({
//   button: {
//     backgroundColor: 'blue',
//     color: 'white',
//     height: 30,
//     lineHeight: 30,
//     marginTop: 10,
//     textAlign: 'center',
//     width: 250
//   },
//   container: {

//   },
//   input: {
//     borderColor: 'black',
//     borderWidth: 1,
//     height: 37,
//     width: 250
//   }
// })

// const submit = values => {
//   console.log('viswanth-submitting form', values)
// }

// const renderInput = ({ input: { onChange, ...restInput }}) => {
//   return <TextInput style={styles.input} onChangeText={onChange} {...restInput} />
// }

// const Form = props => {
//   const { handleSubmit } = props
//   return(
//   <View style={styles.container}>
//   <Text>Email:</Text>
//   <Field name="email" component={renderInput} />
//   <TouchableOpacity onPress={handleSubmit(submit)}>
//     <Text style={styles.button}>Submit</Text>
//   </TouchableOpacity>
// </View>
//   )
// }

// export default reduxForm({
//   form: 'test'
// })(Form)
