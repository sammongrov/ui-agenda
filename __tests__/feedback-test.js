import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Actions } from 'react-native-router-flux';
import FeedbackForm from '../FeedbackForm';

configure({ adapter: new Adapter() });

const initialState = {};
const mockStore = configureStore();

const formProps = {
  handleSubmit: jest.fn(),
  error: false,
  invalid: false,
  submitting: true,
};
configure({ adapter: new Adapter() });
jest.mock('react-native-router-flux', () => ({
  Actions: {
    FeedbackFormScene: jest.fn(),
    pop: jest.fn(),
  },
}));
it('feedback renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={mockStore(initialState)}>
        <FeedbackForm {...formProps} />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('updateSize', () => {
  const height = 720;
  const rootComponent = shallow(<FeedbackForm store={mockStore(initialState)} {...formProps} />)
    .dive()
    .dive()
    .dive();
  const instance = rootComponent.instance();
  instance.updateSize(height);
  expect(rootComponent.state().height).toBe(height);
});
it('onSubmit', () => {
  const values = '';
  const rootComponent = shallow(<FeedbackForm store={mockStore(initialState)} {...formProps} />)
    .dive()
    .dive()
    .dive();
  const instance = rootComponent.instance();
  instance.onSubmit(values);
  expect(rootComponent.values).toBeUndefined();
});

it('Navbar onPress does  calls Actions.pop', () => {
  const root = shallow(<FeedbackForm store={mockStore(initialState)} {...formProps} />)
    .dive()
    .dive()
    .dive();
  Actions.currentScene = 'FeedbackFormScene';
  const button = root
    .find('NavBar')
    .shallow()
    .find('TouchableOpacity');
  button.props().onPress();
  expect(Actions.pop.mock.calls.length).toBe(1);
});
it('Navbar onPress does not  calls Actions.pop', () => {
  Actions.currentScene = 'Agendascene';
  Actions.pop.mockClear();
  const root = shallow(<FeedbackForm store={mockStore(initialState)} {...formProps} />)
    .dive()
    .dive()
    .dive();
  const button = root
    .find('NavBar')
    .shallow()
    .find('TouchableOpacity')
    .first();
  button.props().onPress();
  expect(Actions.pop.mock.calls.length).toBe(0);
});

it('should render the onChange to textName', () => {
  const rootComponent = shallow(<FeedbackForm store={mockStore(initialState)} {...formProps} />)
    .dive()
    .dive()
    .dive();
  const authorInputComponent = rootComponent.find({ name: 'name' }).first();
  authorInputComponent.props().onChange('textName');
  expect(rootComponent.state('name')).toEqual('textName');
});

it('should render the onChange to textMail', () => {
  const rootComponent = shallow(<FeedbackForm store={mockStore(initialState)} {...formProps} />)
    .dive()
    .dive()
    .dive();
  const authorInputComponent = rootComponent.find({ name: 'email' }).first();
  authorInputComponent.props().onChange('textMail');
  expect(rootComponent.state('email')).toEqual('textMail');
});

it('should render the onChange to textComment', () => {
  const rootComponent = shallow(<FeedbackForm store={mockStore(initialState)} {...formProps} />)
    .dive()
    .dive()
    .dive();
  const authorInputComponent = rootComponent.find({ name: 'comment' }).first();
  authorInputComponent.props().onChange('textComment');
  expect(rootComponent.state('comments')).toEqual('textComment');
});

it('calls onContentSizeChange', () => {
  const rootComponent = shallow(<FeedbackForm store={mockStore(initialState)} {...formProps} />)
    .dive()
    .dive()
    .dive();
  const event = { nativeEvent: { contentSize: { height: 720 } } };
  const input = rootComponent.find({ name: 'comment' }).first();
  input.props().onContentSizeChange(event);
  expect(rootComponent.state('height')).toEqual(720);
});
