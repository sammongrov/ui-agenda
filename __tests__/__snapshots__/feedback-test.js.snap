// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`feedback renders correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      4293914607,
      4293914607,
    ]
  }
  locations={
    Array [
      0,
      1,
    ]
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Array [
            undefined,
            null,
          ]
        }
      >
        <View
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#2E88FF",
                    "fontSize": 36,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              ÔÖÅ
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              undefined,
              undefined,
              null,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontFamily": "OpenSans-Regular",
                },
                Array [
                  undefined,
                  null,
                ],
              ]
            }
          >
            Feedback
          </Text>
        </View>
        <View
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <RCTScrollView
        contentContainerStyle={
          Object {
            "backgroundColor": "#F6F7F9",
            "flexGrow": 1,
            "paddingHorizontal": 10,
            "paddingTop": 10,
          }
        }
      >
        <View>
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Name
            </Text>
            <View
              style={
                Array [
                  undefined,
                  false,
                ]
              }
            >
              <TextInput
                allowFontScaling={true}
                autoCapitalize="none"
                autoCorrect={false}
                disableFullscreenUI={true}
                keyboardType="default"
                name="name"
                onBlur={[Function]}
                onChangeText={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                returnKeyType="next"
                style={
                  Array [
                    undefined,
                    undefined,
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Email
            </Text>
            <View
              style={
                Array [
                  undefined,
                  false,
                ]
              }
            >
              <TextInput
                allowFontScaling={true}
                autoCapitalize="none"
                autoCorrect={false}
                disableFullscreenUI={true}
                keyboardType="email-address"
                name="email"
                onBlur={[Function]}
                onChangeText={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                returnKeyType="next"
                style={
                  Array [
                    undefined,
                    undefined,
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Comment
            </Text>
            <View
              style={
                Array [
                  undefined,
                  false,
                ]
              }
            >
              <TextInput
                allowFontScaling={true}
                autoCapitalize="none"
                autoCorrect={false}
                disableFullscreenUI={true}
                keyboardType="default"
                multiline={true}
                name="comment"
                onBlur={[Function]}
                onChangeText={[Function]}
                onContentSizeChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                returnKeyType="next"
                style={
                  Array [
                    undefined,
                    Object {
                      "height": undefined,
                    },
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  undefined,
                ]
              }
            >
              Submit
            </Text>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>
</BVLinearGradient>
`;
